sandbox/sandbox_docker.go

```diff
@@ -148,6 +149,11 @@ func (s *dockerSandbox) Setup(ctx context.Context, config SandboxSetupConfig) er
 		Env:          environmentVariables,
 		Cmd:          s.config.InitCommand,
 		Labels:       s.setup.Labels,
+		// We explicitly set `Entrypoint` to nil to prevent any binary command execution from the base image because we
+		// want to use the base image as a shell environment
+		// When both `Entrypoint` and `Cmd` are provided, the container uses `<Entrypoint> <Cmd>` sequence i.e. `Cmd` is used as args for `Entrypoint`
+		// Eg. https://gitlab.com/safedep/ci-components/vet/-/blob/main/templates/scan.yml?ref_type=heads#L223
+		Entrypoint: nil,
```

Comment:
Hardcoding here is not a good idea. Its like codifying the need for a specific image like `vet` in the generic docker sandbox implementation.

What is the default value? Pointers are supposed to be `nil` by default.

We should introduce a bool or some flag in `SandboxSetupConfig` based on which this is set. Else continue using the default.